include/drivers/peci.h

```diff
@@ -0,0 +1,298 @@
+#define PECI_GET_TEMP_RD_LEN           (2U)
+#define PECI_GET_TEMP_CMD_LEN          (4U)
+#define PECI_GET_TEMP_LSB              (0U)
+#define PECI_GET_TEMP_MSB              (1U)
+#define PECI_GET_TEMP_ERR_MSB          (0x80U)
+#define PECI_GET_TEMP_ERR_LSB_GENERAL  (0x0U)
+#define PECI_GET_TEMP_ERR_LSB_RES      (0x1U)
+#define PECI_GET_TEMP_ERR_LSB_TEMP_LO  (0x2U)
+#define PECI_GET_TEMP_ERR_LSB_TEMP_HI  (0x3U)
+
+/** RdPkgConfig command format. */
+#define PECI_RD_PKG_WR_LEN             (5U)
+#define PECI_RD_PKG_LEN_BYTE           (2U)
+#define PECI_RD_PKG_LEN_WORD           (3U)
+#define PECI_RD_PKG_LEN_DWORD          (5U)
+#define PECI_RD_PKG_CMD_LEN            (8U)
+
+/** WrPkgConfig command format */
+#define PECI_WR_PKG_RD_LEN              (1U)
+#define PECI_WR_PKG_LEN_BYTE            (7U)
+#define PECI_WR_PKG_LEN_WORD            (8U)
+#define PECI_WR_PKG_LEN_DWORD           (10U)
+#define PECI_WR_PKG_CMD_LEN             (9U)
+
+/** RdIAMSR command format */
+#define PECI_RD_IAMSR_WR_LEN            (5U)
+#define PECI_RD_IAMSR_LEN_BYTE          (2U)
+#define PECI_RD_IAMSR_LEN_WORD          (3U)
+#define PECI_RD_IAMSR_LEN_DWORD         (5U)
+#define PECI_RD_IAMSR_LEN_QWORD         (9U)
+#define PECI_RD_IAMSR_CMD_LEN           (8U)
+
+/** WrIAMSR command format */
+#define PECI_WR_IAMSR_RD_LEN            (1U)
+#define PECI_WR_IAMSR_LEN_BYTE          (7U)
+#define PECI_WR_IAMSR_LEN_WORD          (8U)
+#define PECI_WR_IAMSR_LEN_DWORD         (10U)
+#define PECI_WR_IAMSR_LEN_QWORD         (14U)
+#define PECI_WR_IAMSR_CMD_LEN           (9U)
+
+/** RdPCIConfig command format */
+#define PECI_RD_PCICFG_WR_LEN           (6U)
+#define PECI_RD_PCICFG_LEN_BYTE         (2U)
+#define PECI_RD_PCICFG_LEN_WORD         (3U)
+#define PECI_RD_PCICFG_LEN_DWORD        (5U)
+#define PECI_RD_PCICFG_CMD_LEN          (9U)
+
+/** WrPCIConfig command format */
+#define PECI_WR_PCICFG_RD_LEN           (1U)
+#define PECI_WR_PCICFG_LEN_BYTE         (8U)
+#define PECI_WR_PCICFG_LEN_WORD         (9U)
+#define PECI_WR_PCICFG_LEN_DWORD        (11U)
+#define PECI_WR_PCICFG_CMD_LEN          (10U)
+
+/** RdPCIConfigLocal command format */
+#define PECI_RD_PCICFGL_WR_LEN          (5U)
+#define PECI_RD_PCICFGL_RD_LEN_BYTE     (2U)
+#define PECI_RD_PCICFGL_RD_LEN_WORD     (3U)
+#define PECI_RD_PCICFGL_RD_LEN_DWORD    (5U)
+#define PECI_RD_PCICFGL_CMD_LEN         (8U)
+
+/** WrPCIConfigLocal command format */
+#define PECI_WR_PCICFGL_RD_LEN          (1U)
+#define PECI_WR_PCICFGL_WR_LEN_BYTE     (7U)
+#define PECI_WR_PCICFGL_WR_LEN_WORD     (8U)
+#define PECI_WR_PCICFGL_WR_LEN_DWORD    (10U)
+#define PECI_WR_PCICFGL_CMD_LEN         (9U)
+
+/**
+ * @brief PECI buffer structure
+ *
+ * @param buf is a valid pointer on a data buffer, or NULL otherwise.
+ * @param len is the length of the buffer.
+ * @param fcs frame check sequence byte.
+ *        Each message requires at least one fcs.
+ */
+struct peci_buf {
+	u8_t *buf;
+	size_t len;
+	u8_t *fcs;
```

Comment:
How is this validated?

Reply:
fcs removed, this is now validated in peci_handlers as indicated.